.github/workflows/_check-typing.yml

```diff
@@ -0,0 +1,37 @@
+name: Reusable Check Typing
+
+on:
+  workflow_call:
+    inputs:
+      python-version:
+        description: 'Python version to use'
+        default: '3.9'
+        required: false
+        type: string
+
+jobs:
+  mypy:
+    runs-on: ubuntu-20.04
+    steps:
+    - uses: actions/checkout@v3
+
+    - uses: actions/setup-python@v3
+      with:
+        python-version: ${{ inputs.python-version }}
+
+    - name: Cache pip
+      uses: actions/cache@v3
+      with:
+        path: ~/.cache/pip
+        key: ${{ runner.os }}-pip-${{ hashFiles('requirements/pytorch/base.txt') }}
+        restore-keys: |
+          ${{ runner.os }}-pip-
+
+    - name: Install dependencies
+      run: |
+        pip install '.[dev,test]' --extra-index-url https://download.pytorch.org/whl/cpu/torch_stable.html
+        pip install mypy  # don't use --upgrade to respect the version installed via setup.py
+        pip list
+
+    - name: Check typing
+      run: mypy
```

Comment:
how does it know about the config file now? `pyproject.toml`

Reply:
`mypy` looks for a configuration file in this order:

> By default it uses the file mypy.ini with a fallback to .mypy.ini, then pyproject.toml, then setup.cfg in the current directory, then $XDG_CONFIG_HOME/mypy/config, then ~/.config/mypy/config, and finally .mypy.ini in the user home directory if none of them are found; the [--config-file](https://mypy.readthedocs.io/en/stable/command_line.html#cmdoption-mypy-config-file) command-line flag can be used to read a different file instead (see [Config file](https://mypy.readthedocs.io/en/stable/command_line.html#config-file-flag)).

https://mypy.readthedocs.io/en/stable/config_file.html

---

I want this reusable workflow as general as possible so that it can be reused across different repositories regardless of the config file path.